src/components/OrganizationsForm.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  FieldPath,
+  FormProvider,
+  useFieldArray,
+  useForm,
+  useWatch,
+} from "react-hook-form";
+import InputWithStatus from "./InputWithStatus";
+
+const fieldNames = Object.keys(DEFAULT_ORGANIZATION) as Array<
+  keyof Organization
+>;
+
+const OrganizationsForm: React.FC<FormComponentProps> = ({
+  title,
+  labelData,
+  setLabelData,
+}) => {
+  const methods = useForm<LabelData>({
+    defaultValues: labelData,
+  });
+
+  const { control, setValue } = methods;
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "organizations",
+  });
+
+  const watchedOrganizations = useWatch({
+    control,
+    name: "organizations",
+  });
+
+  useEffect(() => {
+    if (watchedOrganizations) {
+      setLabelData((prevLabelData) => ({
+        ...prevLabelData,
+        organizations: watchedOrganizations,
+      }));
+    }
+  }, [watchedOrganizations, setLabelData]);
+
+  const setAllFieldsStatus = useCallback(
+    (orgIndex: number, status: FieldStatus) => {
+      fieldNames.forEach((fieldName) => {
+        const fieldPath =
+          `organizations.${orgIndex}.${fieldName}.status` as FieldPath<LabelData>;
+        setValue(fieldPath, status, {
+          shouldValidate: true,
+          shouldDirty: true,
+        });
+      });
+    },
+    [setValue],
+  );
+
+  const areAllFieldStatus = (index: number, status: FieldStatus) => {
+    const currentOrg = watchedOrganizations?.[index];
+    return checkOrganizationStatus(currentOrg, status);
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <div className="p-4" data-testid="organizations-form">
+        <Typography
+          variant="h6"
+          className="text-lg font-bold"
+          data-testid="form-title"
+        >
+          {title}
+        </Typography>
+        <Box>
+          {fields.map((field, index) => (
+            <Box
+              key={field.id}
+              className="mb-4 py-4"
+              data-testid={`organization-${index}`}
+            >
+              <OrganizationInformation index={index} />
+              <Box className="flex flex-wrap mt-4 justify-end gap-2">
+                <Tooltip
+                  title="Mark all as Verified"
+                  enterDelay={1000}
+                  disableHoverListener={areAllFieldStatus(
+                    index,
+                    FieldStatus.Verified,
+                  )}
+                >
+                  <span>
+                    <Button
+                      variant="outlined"
+                      color="secondary"
+                      onClick={() =>
+                        setAllFieldsStatus(index, FieldStatus.Verified)
+                      }
+                      disabled={areAllFieldStatus(index, FieldStatus.Verified)}
+                      data-testid={`verify-all-btn-${index}`}
+                    >
+                      <DoneAllIcon />
+                    </Button>
+                  </span>
```

Comment:
Shouldn't this be iconButton instead of simple button?

Reply:
With IconButton it looks like this:
![image](https://github.com/user-attachments/assets/9ee5c0bd-53cc-4ec9-9ecf-6560c0c8cd06)

With Button:
![image](https://github.com/user-attachments/assets/aab6569e-008d-443f-a652-b02e0c8974fd)

Please let me know which you prefer. I'm a bit biased towards the squares.